### PR TITLE
Add remaining Decimal SDK functions

### PR DESCRIPTION
**Pull request by alexsieland**
Add the remaining unimplemented functions that are defined in the Morphir spec for Decimal.
The is also a small amount of refactoring, moving a small number of Decimal functions from Native to the DecimalSDK. This is done to ensure all functionality for Decimal is in the same location.

### CODE REVIEW

File: examples/morphir-elm-projects/evaluator-tests/morphir-ir.json3
Comment (DamianReeves) [inline examples/morphir-elm-projects/evaluator-tests/morphir-ir.json3:1]: Is this file intended to be here?

Comment (alexsieland) [reply examples/morphir-elm-projects/evaluator-tests/morphir-ir.json3:1]: It is not. Removed

File: morphir/src/org/finos/morphir/runtime/sdk/DecimalSDK.scala [hunk 1]
```diff
@@ -24,10 +33,16 @@ object DecimalSDK {
       RTDecimal(result)
   }
 
+  val asString = DynamicNativeFunction1("toString") {
```
Comment (michelchan) [inline morphir/src/org/finos/morphir/runtime/sdk/DecimalSDK.scala:36]: the val name should be consistent

Comment (alexsieland) [reply morphir/src/org/finos/morphir/runtime/sdk/DecimalSDK.scala:36]: toString is already defined for all objects in scala. I can do _toString or the likes if we want the name to look the same, but I can't override it, since the return value is different than the existing defined toString

Comment (michelchan) [reply morphir/src/org/finos/morphir/runtime/sdk/DecimalSDK.scala:36]: I forgot about that 😆 let's do `_toString`

**Review by michelchan — APPROVED** [review]
lgtm
